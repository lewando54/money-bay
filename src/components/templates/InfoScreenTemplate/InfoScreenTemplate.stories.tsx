/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import InfoScreenTemplate from './InfoScreenTemplate'

const meta = {
	component: InfoScreenTemplate,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof InfoScreenTemplate>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
