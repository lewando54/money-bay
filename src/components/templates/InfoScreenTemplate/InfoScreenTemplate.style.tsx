import { Dimensions, StyleSheet } from 'react-native'
const { width, height } = Dimensions.get('screen')
import globalThemeStyle from '../../../styling/GlobalTheme.style'
export default StyleSheet.create({
    mainContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    bgImage: {
        width: width,
        height: height,
        resizeMode: 'contain',
        position: 'absolute',
        top: height * -0.1740
    },
    iconTitleContainer: {
        marginBottom: 'auto',
        marginTop: height * 0.2327,
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: width * 0.1233
    },
    titleText: {
        ...globalThemeStyle.text_Bold,
        fontSize: 36
    },
    buttonsContainer: {
        marginTop: 'auto',
        width: '100%',
        padding: width * 0.0533,
        gap: height * 0.0172
    }
})
