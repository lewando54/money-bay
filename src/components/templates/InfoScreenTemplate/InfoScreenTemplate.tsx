import React from 'react'
import { Image, StyleProp, Text, View,ViewStyle } from 'react-native'

import Images from '@assets/images'
import { Button } from '@atoms'
import { colorProp, sizeProp } from 'src/components/atoms/Button/Button'

import InfoScreenTemplateStyle from './InfoScreenTemplate.style'

import { MAIN_DARK } from '@styling/GlobalTheme.style'

interface IInfoScreenTemplateProps {
    icon?: React.ReactNode
    title?: string
    titleColor?: string
    body?: React.ReactNode
    additionalButtons?: Array<{
        children: React.ReactNode,
        color?: colorProp,
        size?: sizeProp,
        width?: string,
        containerStyle?: StyleProp<ViewStyle>,
        onClick?: (() => void) | null | undefined,
        testID?: string
    }>
    onSubmit?: () => void
    testID?: string
}

// 

export default function InfoScreenTemplate({
    icon=<Text>Icon</Text>,
    title='Account created!',
    titleColor=MAIN_DARK,
    body=<Text>Test</Text>,
    additionalButtons,
    onSubmit,
    testID
}: IInfoScreenTemplateProps){
    return (
        <View style={InfoScreenTemplateStyle.mainContainer} testID={testID}>
            <Image source={Images.bgInfoCard} style={InfoScreenTemplateStyle.bgImage}/>
            <View style={InfoScreenTemplateStyle.iconTitleContainer}>
                {icon}
                <Text style={[InfoScreenTemplateStyle.titleText, {color: titleColor}]}>{title}</Text>
                {body}
            </View>
            <View style={InfoScreenTemplateStyle.buttonsContainer}>
                {additionalButtons && additionalButtons.map((item, index) => 
                    <Button 
                        key={index}
                        color={item.color}
                        size={item.size}
                        width={item.width}
                        containerStyle={item.containerStyle}
                        onClick={item.onClick}
                        testID={item.testID}
                    >
                        {item.children}
                    </Button>
                )}
                <Button onClick={onSubmit}>Done</Button>
            </View>
        </View>
    )
}
