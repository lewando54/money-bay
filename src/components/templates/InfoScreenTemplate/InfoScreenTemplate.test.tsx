import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import Images from '@assets/images'

import InfoScreenTemplate from './InfoScreenTemplate'

import globalThemeStyle, { globalColors } from '@styling/GlobalTheme.style'
describe('InfoScreenTemplate', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<InfoScreenTemplate />)
        expect(tree).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const tree = renderer.create(<InfoScreenTemplate 
            icon={<Images.accountSuccessSVG/>}
            title='Account created!'
            titleColor={globalColors.MAIN_DARK}
            body={<Text style={{...globalThemeStyle.text_Regular, color: globalColors.BODY_TEXT_COLOR, fontSize: 16, textAlign: 'center', marginTop: 20}}>Your account has been created successfully.</Text>}
            onSubmit={() => console.log('test')} />)
        expect(tree).toMatchSnapshot()
    })

    it('should render with buttons correctly', () => {
        const tree = renderer.create(<InfoScreenTemplate 
            icon={<Images.accountSuccessSVG/>}
            title='Account created!'
            titleColor={globalColors.MAIN_DARK}
            body={<Text style={{...globalThemeStyle.text_Regular, color: globalColors.BODY_TEXT_COLOR, fontSize: 16, textAlign: 'center', marginTop: 20}}>Your account has been created successfully.</Text>}
            additionalButtons={[{
                children: 'Test',
            }]}
            onSubmit={() => console.log('test')} />)
        expect(tree).toMatchSnapshot()
    })
})
