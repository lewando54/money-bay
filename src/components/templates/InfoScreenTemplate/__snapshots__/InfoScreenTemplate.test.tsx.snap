// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoScreenTemplate should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 1334,
        "position": "absolute",
        "resizeMode": "contain",
        "top": -232.11599999999999,
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "auto",
        "marginTop": 310.42179999999996,
        "paddingHorizontal": 92.47500000000001,
      }
    }
  >
    <Text>
      Icon
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishBold",
            "fontSize": 36,
          },
          {
            "color": "#1B1D4D",
          },
        ]
      }
    >
      Account created!
    </Text>
    <Text>
      Test
    </Text>
  </View>
  <View
    style={
      {
        "gap": 22.9448,
        "marginTop": "auto",
        "padding": 39.975,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4288142079,
              4279654301,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            false,
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ]
          }
        >
          Done
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;

exports[`InfoScreenTemplate should render correctly 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 1334,
        "position": "absolute",
        "resizeMode": "contain",
        "top": -232.11599999999999,
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "auto",
        "marginTop": 310.42179999999996,
        "paddingHorizontal": 92.47500000000001,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishBold",
            "fontSize": 36,
          },
          {
            "color": "#1B1D4D",
          },
        ]
      }
    >
      Account created!
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Your account has been created successfully.
    </Text>
  </View>
  <View
    style={
      {
        "gap": 22.9448,
        "marginTop": "auto",
        "padding": 39.975,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4288142079,
              4279654301,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            false,
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ]
          }
        >
          Done
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;

exports[`InfoScreenTemplate should render with buttons correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 1334,
        "position": "absolute",
        "resizeMode": "contain",
        "top": -232.11599999999999,
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": "auto",
        "marginTop": 310.42179999999996,
        "paddingHorizontal": 92.47500000000001,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        [
          {
            "color": "white",
            "fontFamily": "MulishBold",
            "fontSize": 36,
          },
          {
            "color": "#1B1D4D",
          },
        ]
      }
    >
      Account created!
    </Text>
    <Text
      style={
        {
          "color": "#4C4C60",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Your account has been created successfully.
    </Text>
  </View>
  <View
    style={
      {
        "gap": 22.9448,
        "marginTop": "auto",
        "padding": 39.975,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4288142079,
              4279654301,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            false,
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ]
          }
        >
          Test
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          {
            "colors": [
              4288142079,
              4279654301,
            ],
            "endPoint": [
              1,
              0.5,
            ],
            "locations": undefined,
            "startPoint": [
              0,
              0.5,
            ],
          }
        }
        style={
          [
            {
              "borderRadius": 10,
              "padding": 15,
            },
            false,
            false,
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "MulishSemiBold",
                "fontSize": 16,
                "textAlign": "center",
              },
              {
                "color": "white",
              },
              false,
              false,
            ]
          }
        >
          Done
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
